plugins/parsers/avro/parser.go

```diff
@@ -0,0 +1,238 @@
+package avro
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/parsers"
+	"github.com/jeremywohl/flatten"
+	"github.com/linkedin/goavro/v2"
+	"time"
+)
+
+type Parser struct {
+	MetricName      string   `toml:"metric_name"`
+	SchemaRegistry  string   `toml:"avro_schema_registry"`
+	Schema          string   `toml:"avro_schema"`
+	Measurement     string   `toml:"avro_measurement"`
+	Tags            []string `toml:"avro_tags"`
+	Fields          []string `toml:"avro_fields"`
+	Timestamp       string   `toml:"avro_timestamp"`
+	TimestampFormat string   `toml:"avro_timestamp_format"`
+	DiscardArrays   bool     `toml:"avro_discard_arrays"`
+	FieldSeparator  string   `toml:"avro_field_separator"`
+	DefaultTags     map[string]string
```

Comment:
Please change to
```suggestion
	DefaultTags     map[string]string `toml:"-"`
```
to prevent setting this through the configuration.